database/node.go

```diff
@@ -32,66 +33,97 @@ const (
 // WithAddressOption. It is used in tests to override address detection.
 var DefaultBindAddress = ""
 
-// OptionFactory creates Dqlite `App` initialisation arguments and options.
-// These generally depend on a controller's agent config.
-type OptionFactory struct {
+// NodeManager is responsible for interrogating a single Dqlite node,
+// and emitting configuration for starting its Dqlite `App` based on
+// operational requirements and controller agent config.
+type NodeManager struct {
 	cfg    agent.Config
 	port   int
 	logger Logger
 
+	dataDir     string
 	bindAddress string
 }
 
-// NewOptionFactory returns a new OptionFactory reference
+// NewNodeManager returns a new NodeManager reference
 // based on the input agent configuration.
-func NewOptionFactory(cfg agent.Config, logger Logger) *OptionFactory {
-	return &OptionFactory{
+func NewNodeManager(cfg agent.Config, logger Logger) *NodeManager {
+	return &NodeManager{
 		cfg:         cfg,
 		port:        dqlitePort,
 		logger:      logger,
 		bindAddress: DefaultBindAddress,
 	}
 }
 
+// IsExistingNode returns true if this machine or container has
+// ever started a Dqlite `App` before. Specifically, this is whether
+// the Dqlite data directory is empty.
+func (m *NodeManager) IsExistingNode() (bool, error) {
+	if _, err := m.EnsureDataDir(); err != nil {
+		return false, errors.Annotate(err, "ensuring Dqlite data directory")
+	}
+
+	dir, err := os.Open(m.dataDir)
+	if err != nil {
+		return false, errors.Annotate(err, "opening Dqlite data directory")
+	}
+
+	_, err = dir.Readdirnames(1)
```

Comment:
We might actually want to know what files are in this directory at some point, for now, this seems sensible.